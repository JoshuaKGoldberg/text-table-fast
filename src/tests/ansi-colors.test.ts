import color from "cli-color";
import { describe, expect, test } from "vitest";

import { textTable } from "../index.js";

describe("ansi-colors", () => {
	test.each([
		[[[color.red("Red"), color.green("Green"), color.blue("Blue")]]],
		[
			[
				[color.red("Red"), color.green("Green"), color.blue("Blue")],
				[
					color.bold("Bold"),
					color.underline("Underline"),
					color.italic("Italic"),
				],
			],
		],
		[
			[
				[color.red("Red"), color.green("Green"), color.blue("Blue")],
				[
					color.bold("Bold"),
					color.underline("Underline"),
					color.italic("Italic"),
				],
				[
					color.inverse("Inverse"),
					color.strike("Strike"),
					color.blink("Blink"),
				],
				["bar", "45", "abcd"],
			],
		],
	])("%j", (rows) => {
		expect(
			"\n" +
				color.strip(
					textTable(rows, {
						align: ["left", "center", "left"],
						stringLength: (value) => color.strip(value).length,
					}),
				) +
				"\n",
		).toMatchSnapshot();
	});
});
