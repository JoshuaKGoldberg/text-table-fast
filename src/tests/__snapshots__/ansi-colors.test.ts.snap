// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ansi-colors > [["\\u001b[31mRed\\u001b[39m","\\u001b[32mGreen\\u001b[39m","\\u001b[34mBlue\\u001b[39m"],["\\u001b[1mBold\\u001b[22m","\\u001b[4mUnderline\\u001b[24m","\\u001b[3mItalic\\u001b[23m"],["\\u001b[7mInverse\\u001b[27m","\\u001b[9mStrike\\u001b[29m","\\u001b[5mBlink\\u001b[25m"],["bar","45","abcd"]] 1`] = `
"
Red        Green    Blue
Bold     Underline  Italic
Inverse    Strike   Blink
bar          45     abcd
"
`;

exports[`ansi-colors > [["\\u001b[31mRed\\u001b[39m","\\u001b[32mGreen\\u001b[39m","\\u001b[34mBlue\\u001b[39m"],["\\u001b[1mBold\\u001b[22m","\\u001b[4mUnderline\\u001b[24m","\\u001b[3mItalic\\u001b[23m"]] 1`] = `
"
Red     Green    Blue
Bold  Underline  Italic
"
`;

exports[`ansi-colors > [["\\u001b[31mRed\\u001b[39m","\\u001b[32mGreen\\u001b[39m","\\u001b[34mBlue\\u001b[39m"]] 1`] = `
"
Red  Green  Blue
"
`;
